<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>

<body>
  <script>
    /* const promise = new Promise((resolve, reject) => {
      resolve("success1");
      reject("error");
      resolve("success2");
    });
    promise
      .then(res => {
        console.log("then: ", res);
      }).catch(err => {
        console.log("catch: ", err);
      }) */


    /* const promise = new Promise((resolve, reject) => {
      reject("error");
      resolve("success2");
    });
    promise
      .then(res => {
        console.log("then1: ", res);
      }).then(res => {
        console.log("then2: ", res);
      }).catch(err => {
        console.log("catch: ", err);
      }).then(res => {
        console.log("then3: ", res);
      }) */


    /* Promise.resolve(1)
      .then(res => {
        console.log(res);
        return 2;
      })
      .catch(err => {
        return 3;
      })
      .then(res => {
        console.log(res);
      }); */


    /* Promise.reject(1)
      .then(res => {
        console.log(res);
        return 2;
      })
      .catch(err => {
        console.log(err);
        return 3
      })
      .then(res => {
        console.log(res);
      }); */


    /* Promise.reject(1)
      .then(res => {
        console.log(res);
        return 2;
      })
      .catch(err => {
        console.log(err);
        return 3
      })
      .then(res => {
        console.log(res);
      }); */


    /* const promise = new Promise((resolve, reject) => {
      setTimeout(() => {
        console.log('timer')
        resolve('success')
      }, 1000)
    })
    const start = Date.now();
    promise.then(res => {
      console.log(res, Date.now() - start)
    })
    promise.then(res => {
      console.log(res, Date.now() - start)
    }) */


    /* Promise.resolve().then(() => {
      return new Error('error!!!')
    }).then(res => {
      console.log("then: ", res)
    }).catch(err => {
      console.log("catch: ", err)
    }) */


    const promise = Promise.resolve().then(() => {
      return promise;
    })
    promise.catch(console.err)








  </script>
</body>

</html>